### PR TITLE
Additional features and CR Changes

### PR DESCRIPTION
**Pull request by jyoung488**
Changes from PR #79 and suggestions from Nate
Adding in these features:
- Adjusting the short circuit methods
- #72: Signer Groups
- Adjusting unit tests

### CODE REVIEW

**Comment by nathanbuchar** [comment]
I think we should still test for PHP 5.5 support. I'm sure not all of our users are running PHP 7, and we're not explicitly using and new PHP 7 syntax like Array literals or anonymous functions (as far as I know), so if we can passively support older PHP versions then I think we should

File: library/HelloSign/Test/SignatureRequestTest.php [hunk 1]
```diff
@@ -229,16 +229,17 @@ public function testSendSignatureRequestWithSignerGroup()
...
         $request->setMessage("Sign this NDA and then we can discuss more. Let me know if you have any questions.");
         $request->addFile(__DIR__ . '/nda.docx');
 
         // Add Signer Group to Signature Request
         $request->addGroup("Authorized Signatory", 0);
-        $request->addGroupSigner("Possible Signer 1", "jen.young+1@hellosign.com", 0);
-        $request->addGroupSigner("Possible Signer 2", "jen.young+2@hellosign.com", 1);
+        $request->addGroupSigner("Jack Example", "jack@example.com", 0);
+        $request->addGroupSigner("Jill Example", "jill@example.com", 1);
+        $request->addGroupSigner("Jane Example", "jane@example.com", 2);
```
Comment (nathanbuchar) [inline library/HelloSign/Test/SignatureRequestTest.php:242]: +1 for using fake email addresses

File: library/HelloSign/AbstractSignatureRequest.php [hunk 2]
```diff
@@ -173,15 +173,16 @@ public function addGroup($name, $group_index_or_role = 0)
     {
       $group = new SignerGroup(array(
         'name' => $name
-      ));
+        )
+      );
```
Comment (nathanbuchar) [inline library/HelloSign/AbstractSignatureRequest.php]: Incorrect indentation

```diff
  $group = new SignerGroup(array(
    'name' => $name
-   )
- );
+ ));
```

**Comment by nathanbuchar** [comment]
Probably just a refactoring oversight, but when you cleaned up the short-circuiting logic, you left semicolons at the end of the conditionals:

```php
if ($foo) {
  return $bar;
}; // <- unnecessary semicolon
```

**Comment by jyoung488** [comment]
I added v5.5 back in but the Travis tests are failing due to dependencies that require v7.1

**Comment by nathanbuchar** [comment]
@jyoung488 Ah, so we do have PHP 7 dependencies? In that case, this lib won't work on PHP <7. Is that correct?

**Comment by jyoung488** [comment]
@nathanbuchar yeah :( there are several dependencies requiring ^7.1

**Comment by jyoung488** [comment]
Also removed the unnecessary semicolons! Thanks for that note - still learning the language so this is super helpful

**Comment by nathanbuchar** [comment]
@jyoung488 If so, then yeah let's remove PHP 5.5 from Travis CI. We also need to update the README, as it currently says:

> ### Requirements
> The latest version of the SDK requires PHP version 5.5 or higher.

**Review by nathanbuchar — CHANGES_REQUESTED** [review]

**Comment by jyoung488** [comment]
@nathanbuchar Thanks! Made these changes.

**Review by nathanbuchar — APPROVED** [review]

**Pull request merged**